services/smartcards/smartcards/core.py

```diff
@@ -23,8 +23,12 @@ def card_reader():
     is_connected = CardInterface.is_reader_connected()
     return json.dumps({"connected": is_connected})
 
+
 @app.route('/card/read')
 def card_read():
+    if CardInterface.has_connection_error():
+        return json.dumps({"present": True, "connectionError": True})
```

Comment:
I'm wondering if we might want to not set `present` to `True` in this case, and constrain the semantics of that field to be "there's a readable card present", not just "there's a piece of plastic present.

Reply:
Check out the relevant types on the client side: https://github.com/votingworks/vxsuite/blob/4d3dbb213d24c2f362ab280f027701c225dcdc33/libs/utils/src/types.ts#L100-L166

I'd probably aim toward what @benadida said and use `present: false` with another optional field that indicates a card connection error.

Reply:
Ok thanks, that makes sense.

I considered adding a `status` field to the API to indicate the disjoint states a card could be in (e.g. `present`, `no_card`, `error`), but decided that it would probably be a large change for consumers of the API. But it's worth noting that now we have one of those weird APIs where there are multiple flags that can only pair up in certain ways. Just something to think about for API design in the future - using a flag is not the most extensible strategy.